packages/keyring/src/pair/index.spec.ts

```diff
@@ -234,6 +235,51 @@ describe('pair', (): void => {
     ).toThrow('Cannot sign with a locked key pair');
   });
 
+  it('allows encrypt/decrypt with ed25519 keypair', (): void => {
+    const message = new Uint8Array([0x61, 0x62, 0x63, 0x64, 0x65]);
+    const encryptedMessage = keyring.alice.encrypt(message, keyring.bob.publicKey);
+
+    expect(keyring.bob.decrypt(encryptedMessage)).toEqual(message);
+    expect(keyring.alice.decrypt(encryptedMessage)).toEqual(null);
+  });
+
+  it('allows encrypt/decrypt with sr25519 keypair', (): void => {
+    const aliceSR25519KeyPair = new Keyring().createFromUri(mnemonicGenerate(), { name: 'sr25519 pair' }, 'sr25519');
+    const bobSR25519KeyPair = new Keyring().createFromUri(mnemonicGenerate(), { name: 'sr25519 pair' }, 'sr25519');
+    const message = new Uint8Array([0x61, 0x62, 0x63, 0x64, 0x65]);
+    const encryptedMessage = aliceSR25519KeyPair.encrypt(message, bobSR25519KeyPair.publicKey);
+
+    expect(bobSR25519KeyPair.decrypt(encryptedMessage)).toEqual(message);
+    expect((): Uint8Array | null => aliceSR25519KeyPair.decrypt(encryptedMessage)).toThrow("Mac values don't match");
+  });
```

Comment:
We would need an additional test (on the back of the above above comment), where the one party uses ed25519 and the other party using sr25519 and they can exchange messages bi-directionally.

Reply:
This test has not been done as you cannot exchange messages if the 2 key types differs.